sys/arm64/arm64/minidump_machdep.c

```diff
@@ -127,8 +127,13 @@ blk_write(struct dumperinfo *di, char *ptr, vm_paddr_t pa, size_t sz)
 			ptr += len;
 			sz -= len;
 		} else {
-			dump_va = (void *)PHYS_TO_DMAP(pa);
 			fragsz += len;
+			/*
+			 * Note that this is safe because all writes using pa
+			 * are at most PAGE_SIZE.
+			 */
+			dump_va = (void *)cheri_kern_setboundsexact(
+			    PHYS_TO_DMAP(pa), fragsz);
```

Comment:
Here I'd say len makes more sense given that's what we modify pa and sz by, but fragsz makes things more confusing, especially as that's what blk_flush uses. AFAICT fragsz is always going to be the same as len given blk_flush resets it back to 0, though amd64 is more complicated, and confuses me since dump_va will always point at the last page...

Reply:
I agree, will change this by using `len`